pkg/github/context_tools.go

```diff
@@ -90,3 +92,86 @@ func GetMe(getClient GetClientFn, t translations.TranslationHelperFunc) (mcp.Too
 
 	return tool, handler
 }
+
+func GetTeams(getClient GetClientFn, getGQLClient GetGQLClientFn, t translations.TranslationHelperFunc) (mcp.Tool, server.ToolHandlerFunc) {
+	tool := mcp.NewTool("get_teams",
+		mcp.WithDescription(t("TOOL_GET_TEAMS_DESCRIPTION", "Get details of the teams the user is a member of")),
+		mcp.WithString("user",
+			mcp.Description(t("TOOL_GET_TEAMS_USER_DESCRIPTION", "Username to get teams for. If not provided, uses the authenticated user.")),
+		),
+		mcp.WithToolAnnotation(mcp.ToolAnnotation{
+			Title:        t("TOOL_GET_TEAMS_TITLE", "Get teams"),
+			ReadOnlyHint: ToBoolPtr(true),
+		}),
+	)
+
+	type args struct {
+		User *string `json:"user,omitempty"`
+	}
+	handler := mcp.NewTypedToolHandler(func(ctx context.Context, _ mcp.CallToolRequest, a args) (*mcp.CallToolResult, error) {
+		var username string
+		if a.User != nil && *a.User != "" {
+			username = *a.User
+		} else {
+			client, err := getClient(ctx)
+			if err != nil {
+				return mcp.NewToolResultErrorFromErr("failed to get GitHub client", err), nil
+			}
+
+			user, res, err := client.Users.Get(ctx, "")
+			if err != nil {
+				return ghErrors.NewGitHubAPIErrorResponse(ctx,
+					"failed to get user",
+					res,
+					err,
+				), nil
+			}
+			username = user.GetLogin()
+		}
+
+		gqlClient, err := getGQLClient(ctx)
+		if err != nil {
+			return mcp.NewToolResultError(fmt.Sprintf("failed to get GitHub GQL client: %v", err)), nil
+		}
+
+		var q struct {
+			User struct {
+				Organizations struct {
+					Nodes []struct {
+						Login githubv4.String
+						Teams struct {
+							Nodes []struct {
+								Name        githubv4.String
+								Slug        githubv4.String
+								Description githubv4.String
+							}
+						} `graphql:"teams(first: 100, userLogins: [$login])"`
+					}
+				} `graphql:"organizations(first: 100)"`
+			} `graphql:"user(login: $login)"`
+		}
+		vars := map[string]interface{}{
+			"login": githubv4.String(username),
+		}
+		if err := gqlClient.Query(ctx, &q, vars); err != nil {
+			return mcp.NewToolResultError(err.Error()), nil
+		}
+
+		t := q.User.Organizations.Nodes
+		if len(t) == 0 {
+			return mcp.NewToolResultError("no teams found for user"), nil
```

Comment:
```suggestion
			return mcp.NewToolResultError("no orgs found for user"), nil
```
Alternatively, just return the results with an empty list of orgs/teams. Our other graphQL tools don't usually give a special message for empty results. (e.g ListDiscussions).